### PR TITLE
Add regex to grammar

### PR DESCRIPTION
**Pull request by beckyjackson**
This allows regex expressions in Perl format to be used in functions:
```
[operator]/regex/[flags]
```

... and
```
[operator]/regex/replacement/[flags]
```

The operator & flags are optional. The delimiter is always a forward slash.

I had to do some hacky things to allow escaped forward slashes to be used within the patterns and replacements, but it seems to work for all cases.

I wanted to just use negative lookbehind, but Nearley doesn't allow that.

There are two problems with this currently. The first one I think we can work around:
1. Escape characters get doubled in the output, e.g., `pat\/ern` becomes `pat\\/ern`. I tried doing a replace, but I think this is actually happening when it gets printed out. We should test this in Python and see if it only includes the one escape character.
2. You cannot use a pattern with an escaped slash at the very end for matching (it's OK in substitution). It looks like the grammar gets confused between a match and a substitution here and returns two results:
```
$ nearley-test -q -i 'regex(s/pattern\//g)'  build/valve_grammar.js
[ [ { type: 'function',
      name: 'regex',
      args:
       [ { type: 'regex',
           operator: 's',
           pattern: 'pattern\\',
           replace: '',
           flags: 'g' } ] } ],
  [ { type: 'function',
      name: 'regex',
      args:
       [ { type: 'regex',
           operator: 's',
           pattern: 'pattern\\/',
           flags: 'g' } ] } ] ]
```

@jamesaoverton could you let me know what you think? Currently we have some regex stuff in `valve.py` that I just parse in Python, but I think it would be cool to do this in the grammar.

### CODE REVIEW

**Comment by jamesaoverton** [comment]
Looks good.

I think we only want the `s` operator. If we require that, then that would distinguish the trailing slash case from an empty substitution. Do you want to do it that way?

**Comment by beckyjackson** [comment]
Yes! That perfectly solves the problem 😄

File: Makefile [hunk 1]
```diff
@@ -17,6 +17,8 @@ build/actual.txt: build/valve_grammar.js | build
 	nearley-test -q -i 'split(prefix.prefix, "&", foo(bar), CURIE(prefix.prefix))' $< >> $@
 	nearley-test -q -i 'a(b(c(d)))' $< >> $@
 	nearley-test -q -i 'in(with-dash."space column")' $< >> $@
+	nearley-test -q -i 'regex(s/pattern/replacement/g)' $< >> $@
+	nearley-test -q -i 'regex(/pat\/ern/replacement/)' $< >> $@
```
Comment (jamesaoverton) [inline Makefile]: This test fails.

**Review by jamesaoverton — APPROVED** [review]
The implementation is good. Please fix the tests and merge.

**Pull request merged**